### ISSUE TITLE
torch.roll has arguments dims for both dim/multidim while other functions have argument dim for the same usage

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#68616]
Inconsistent wrt other functions...

**Comment by ngimel** [comment]
You are right, that's an inconsistency, however, due to bc concerns we are unlikely to fix that.

**Comment by vadimkantorov** [comment]
`dim` argument could be added in addition to dims as an alias, and dims could be hidden from docs if bc is concern

**Comment by vadimkantorov** [comment]
What's also bad is that `dims` is second argument, so we can write tensor.flip(-1), but can't write tensor.roll(-1, shifts = 1)

**Comment by vadimkantorov** [comment]
with my solution, older programs will continue to work, so it should not break bc-compat (or is forward compat more correct term here?)

**Comment by ezyang** [comment]
Reclassified. Internal support for "alias" argument names is not very good but probably worth adding.

**Comment by vadimkantorov** [comment]
same for `torch.tile` and `torch.flip`

**Comment by vadimkantorov** [comment]
Similar problems with `dims` arg:
- https://github.com/pytorch/pytorch/issues/18095
- https://github.com/pytorch/pytorch/issues/106959

**Comment by vadimkantorov** [comment]
Also, docs are not clear that `torch.rand(3,4,5).flip(-1)` work as well, e.g. a single integer is accepted for the `dims` argument:
![image](https://github.com/user-attachments/assets/fea04e2d-1f4e-418d-a930-4b82c1f462d3)
Should likely be:
dims (a list or tuple or int) – axis to flip on`
